src/passes/OptimizeInstructions.cpp

```diff
@@ -1292,6 +1292,23 @@ struct OptimizeInstructions
   }
 
   template<typename T> Expression* optimizePowerOf2FDiv(Binary* binary, T c) {
+    //
+    // x / C_pot    =>   x * (C_pot ^ -1)
+    //
+    // Explanation:
+    // Float point format represent as:
+    //    ((-1) ^ sign) * (2 ^ (exp - bias)) * (1 + significand)
+    //
+    // If we have pow of two numbers than mantissa (significand) have all zeros.
+    // And let's focused on exponent ignoring sign part:
+    //    (2 ^ (exp - bias))
+    //
+    // and for inverted power of two float point:
+    //     1.0 / (2 ^ (exp - bias))   ->   2 ^ -(exp - bias)
+    //
+    // So inversion of C_pot is valid due to it change only sign of exponent
+    // part and don't touch significand part which still remain
+    // the same (zeros).
```

Comment:
Awesome! This really helps make the transformation clear. I made some spelling and grammar fixes, but you might have to fix up the line lengths.

```suggestion
    // Floating point numbers are represented as:
    //    ((-1) ^ sign) * (2 ^ (exp - bias)) * (1 + significand)
    //
    // If we have power of two numbers, then the mantissa (significand) is all zeros.
    // Let's focus on the exponent, ignoring the sign part:
    //    (2 ^ (exp - bias))
    //
    // and for inverted power of two floating point:
    //     1.0 / (2 ^ (exp - bias))   ->   2 ^ -(exp - bias)
    //
    // So inversion of C_pot is valid because it changes only the sign of the exponent
    // part and doesn't touch the significand part, which remains the same (zeros).
```